static/index.js

```diff
@@ -27,48 +45,81 @@ function register() {
         body: JSON.stringify({
           endpoint: subscription.endpoint,
           key: key ? btoa(String.fromCharCode.apply(null, new Uint8Array(key))) : '',
+          machineId: machineId,
+          token: domTokenInput.value,
+          name: domMachineName.value
         }),
       }).then(function(response) {
         response.text().then(function(token) {
-          localforage.setItem('token', token);
-          document.getElementById('token').textContent = token;
-          document.getElementById('registrationForm').style.display = 'none';
-          document.getElementById('unregistrationForm').style.display = 'block';
+          if (response.ok) {
+              localforage.setItem('token', token);
+              domToken.textContent = token;
+              showSection('unregistrationForm');
+          } else {
+            alert('Error: ' + token);
+          }
         });
       });
     });
   });
 }
 
-document.getElementById('register').onclick = register;
+domRegister.onclick = register;
+
+var sections = ['registrationForm', 'unregistrationForm'];
+function showSection(section) {
+  for (var index = 0; index < sections.length; index++) {
+    if (sections[index] === section) {
+      document.getElementById(section).style.display = 'block';
+    } else {
+      document.getElementById(sections[index]).style.display = 'none';
+    }
+  }
+}
 
-document.getElementById('unregister').onclick = function() {
+domUnregister.onclick = function() {
   localforage.getItem('token').then(function(token) {
-    fetch('./unregister', {
+    fetch('./unregisterMachine', {
       method: 'post',
       headers: {
         'Content-type': 'application/json'
       },
       body: JSON.stringify({
         token: token,
+        machineId: machineId
       }),
     }).then(function(response) {
-      document.getElementById('registrationForm').style.display = 'block';
-      document.getElementById('unregistrationForm').style.display = 'none';
-      document.getElementById('token').textContent = '';
-
-      localforage.clear();
+      domToken.textContent = '';
+      showSection('registrationForm');
+      localforage.removeItem('token');
     });
   });
 };
 
+function makeId(length) {
```

Comment:
You could use: https://developer.mozilla.org/en-US/docs/Web/API/RandomSource/getRandomValues#Example